### PR TITLE
do not inherit internal class in public RoadLabelSurfaceLayer

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Refs https://github.com/mapbox/mapbox-navigation-android/pull/6541#discussion_r1012168842.

### CODE REVIEW

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt [hunk 1]
```diff
@@ -31,8 +31,16 @@ import com.mapbox.navigation.ui.shield.model.RouteShield
  * removing the listener with [MapboxCarMap.unregisterObserver].
  */
 @OptIn(MapboxExperimental::class)
-class RoadLabelSurfaceLayer(
-    val carContext: CarContext,
+class RoadLabelSurfaceLayer internal constructor(
+    delegate: RoadLabelSurfaceLayerDelegate,
+) : MapboxCarMapObserver by delegate {
+
+    constructor(carContext: CarContext) : this(RoadLabelSurfaceLayerDelegate(carContext))
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:38]: the carContext is in MapboxCarMap (get it from the onAttached), doesn't have to be this change but we'll want to remove it from the constructor

**Review by kmadsen — APPROVED** [review]

**Comment by Zayankovsky** [comment]
This is actually wrong, because `RoadLabelSurfaceLayerDelegate.children` is not called, therefore road label widget is not drawn.

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt [hunk 2]
```diff
@@ -31,25 +31,30 @@ import com.mapbox.navigation.ui.shield.model.RouteShield
...
+    constructor() : this(RoadLabelSurfaceLayerDelegate())
+
+    override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
+        delegate.onAttached(mapboxCarMapSurface)
+    }
+
+    override fun onDetached(mapboxCarMapSurface: MapboxCarMapSurface) {
+        delegate.onDetached(mapboxCarMapSurface)
+    }
+}
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/roadlabel/RoadLabelSurfaceLayer.kt:47]:
```
class RoadLabelSurfaceLayer : MapboxCarMapObserver {

    var delegate: RoadLabelSurfaceLayerDelegate? = null

    override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
        delegate = RoadLabelSurfaceLayerDelegate(mapboxCarMapSurface.carContext)
        delegate.onAttached(mapboxCarMapSurface)
    }

    override fun onDetached(mapboxCarMapSurface: MapboxCarMapSurface) {
        delegate.onDetached(mapboxCarMapSurface)
        delegate = null
    }
}
```

**Review by kmadsen — APPROVED** [review]
It's better!

**Comment by Zayankovsky** [comment]
> This is actually wrong, because `RoadLabelSurfaceLayerDelegate.children` is not called, therefore road label widget is not drawn.

Looks like the only thing that I missed was to add delegation for `onVisibleAreaChanged`, because `children` is only called from `onAttached`, `onDetached` and `onVisibleAreaChanged`.

**Pull request merged**